src/views/Dog/Dog.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { fetchDogById } from '../../services/dog';
+import { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom/cjs/react-router-dom.min';
+import DogDetails from '../../components/DogDetails/DogDetails';
+import './Dog.css';
+
+export default function Dog() {
+  const [dog, setDog] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const params = useParams();
+
+  useEffect(() => {
+    let timer;
+    const fetchData = async () => {
+      const data = await fetchDogById(params.id);
+      setDog(data);
+      timer = setTimeout(() => {
+        setLoading(false);
+      }, 10);
+    };
+    if (loading) {
```

Comment:
No need to have this fetch function dependent upon loading. It should run once when the component first mounts